skimage/filters/rank/generic.py

```diff
@@ -142,9 +142,9 @@ def _preprocess_input(image, footprint=None, out=None, mask=None,
         n_bins = int(max(3, image.max())) + 1
 
     if n_bins > 2 ** 10:
-        warn("Bad rank filter performance is expected due to a "
-             "large number of bins ({}), equivalent to an approximate "
-             "bitdepth of {:.1f}.".format(n_bins, np.log2(n_bins)),
+        warn(f"Bad rank filter performance is expected due to a "
+             f"large number of bins ({n_bins}), equivalent to an approximate "
+             f"bitdepth of {np.log2(n_bins):.1f}.",
```

Comment:
Not strictly related to this PR, but we might as well replace double quotes with single quotes, since double quotes are note necessary here ;)

Reply:
@mkcor, did you see a quotation style recommended somewhere in our docs? Unfortunately, I think the library is pretty inconsistent as-is on single vs. double quotes. It does seem that single-quotes would be more consistent with most strings in this file, though, so I am not against it.

Reply:
@grlee77 my reference is this discussion with @jni https://github.com/scikit-image/scikit-image/pull/4850#discussion_r461272861 :wink:

Reply:
Makes sense @mkcor and @grlee77 - thanks for that. I personally lean towards single quotes as well and agree that some consistency here would be nice 😄 I went ahead and converted all double quoted strings to single quotes in the files initially affected by this PR, and at least replaced the double quoted version in the deprecation decorator within the filters package, which had been copy + pasted a bit.

I am hoping to contribute to this project quite a bit - will make sure to try to work on standardizing on single quotes in other parts of the code as I interact with them in the future.

Reply:
> I am hoping to contribute to this project quite a bit

Wonderful!

> will make sure to try to work on standardizing on single quotes in other parts of the code as I interact with them in the future.

Don't hesitate to split your contributions into several commits and even several PRs! For example, ba04cac27daaf53599e738086e7aba26170d9810 could/should typically consist of

* one commit for replacing double quotes with single quotes in f-strings;
* one commit submitted through another PR for replacing double quotes with single quotes in the rest of the subpackage (just because this PR is about "Replacing filters .format() calls with f-strings").

No big deal, it's just that sprinting time is mentoring time, right? :wink:

Reply:
@alex-jw-brooks the rationale being that it's easier to review a PR when its commits are short, self-consistent, and well-organized. A commit should correspond to one given conceptual entity.

Reply:
Sure that's fair, I'm all for atomic commits and manageable PRs that make sense in retrospect.

In general, does `scikit-image` expect a 1:1 correspondence between PRs and issues, and if so, is it permissible that I create issues when working on things like style changes for quotes, or is there a process that I should follow to get that on the board? Further, would you like me to split this PR up to only include changes on f-strings, and in the quote standardization PR that would follow, just go ahead and do the rest of the `filters` package?

Reply:
> In general, does scikit-image expect a 1:1 correspondence between PRs and issues

Not particularly: You can submit PRs 'spontaneously' and, sometimes, reported issues get addressed by more than one PR.

> is it permissible that I create issues when working on things like style changes for quotes

Of course! Feel free to create new issues.

> or is there a process that I should follow to get that sort of thing on the board when it isn't already

When contributions are not ready, don't hesitate to create PRs as **draft** PRs. This way, reviewers will know that the work isn't complete yet. But you might still want to discuss certain things or get feedback on your work in progress.

> Further, would you like me to split this PR up to only include changes on f-strings, and in the quote standardization PR that would follow, just go ahead and do the rest of the filters package?

I would prefer it if this PR only contained the changes for f-strings; also because it's the weekend, so people are not available to chime in and weigh in, should they have strong opinions on single/double quotes. If some people much preferred double quotes, I would feel bad forcing so many single quotes at once on them... And, just out of consistency: We said we were taking care of f-strings in the `filters` subpackage :relieved: